core/types/calls.cc

```diff
@@ -696,7 +696,19 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, DispatchArgs args, core
     // the send, assume that the last argument is an implicit keyword args hash.
     bool implicitKwsplat = false;
     if (ait != aPosEnd && hasKwargs && args.args.size() == args.numPosArgs) {
-        // NOTE: this would be a good place for an autocorrect to using `**kwhash`
+        auto splatLoc = core::Loc(args.locs.file, args.locs.args[args.args.size() - 1]);
+
+        // If --error-kwarg-hash-without-splat is set, we will treat "**-less" keyword hash argument as an error.
+        if (gs.errorKwargHashWithoutSplat) {
+            if (auto e = gs.beginError(splatLoc, errors::Infer::KeywordArgHashWithoutSplat)) {
+                e.setHeader("(`{}`) Supplying a keyword argument hash as the last "
+                            "positional argument without using `{}` is deprecated "
+                            "in Ruby 2.7, and will be an error in Ruby 3.0",
+                            "--error-kwarg-hash-without-splat", "**");
```

Comment:
This error looks a little bit long, how does it look on an 80-col terminal? You could also use `e.addErrorSection` to provide more information below the actual error.

Reply:
I split this up into a few lines. Still doesn't quite _fit_ on an 80-column terminal, but won't look as bad as this was. What do you think?

```
badsplat.rb:7: Keyword argument hash without ** is deprecated https://srb.help/7033
     7 |f(6, someArgs)
             ^^^^^^^^
    badsplat.rb:7: This produces a runtime warning in Ruby 2.7, and will be an error in Ruby 3.0
     7 |f(6, someArgs)
             ^^^^^^^^
    badsplat.rb:7: Note: disabling --error-kwarg-hash-without-splat will suppress this error
     7 |f(6, someArgs)
             ^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    badsplat.rb:7: Replace with **someArgs
     7 |f(6, someArgs)
             ^^^^^^^^
Errors: 1
```